build_tools/ci/cpu_comparison/output_comparer.py

```diff
@@ -2,65 +2,147 @@
+        nb_discrepencies_to_show = min(max_discrepancies_to_show, self.nb_not_close)
+        summary = f"\n\nDiscrepencies at first {nb_discrepencies_to_show} indices:\n\n"
+        coords = np.array(self.where_not_close).T[0:nb_discrepencies_to_show]
+        coords = str(coords).split("\n")
+        for i in range(nb_discrepencies_to_show):
+            if i != nb_discrepencies_to_show - 1:
+                coords[i] += " "
+            coords[i] = (
+                coords[i]
+                + " | "
+                + "{: <23}".format(self.baseline_values_where_not_close[i])
+                + " | "
+                + "{: <23}".format(self.aie_values_where_not_close[i])
+            )
+        indices = [i for i, c in enumerate(coords[0]) if c == "|"]
+        assert len(indices) == 2
+        barrier_0 = indices[0]
+        barrier_1 = indices[1]
+        header_line = " Index"
+        header_line = header_line + " " * (barrier_0 - len(header_line))
+        header_line = header_line + "| Baseline"
+        header_line = header_line + " " * (barrier_1 - len(header_line))
+        header_line = header_line + "| AIE"
+        coords.insert(0, header_line)
+        underline = ["-" for i in range(len(header_line))]
+        underline[barrier_0] = "+"
+        underline[barrier_1] = "+"
+        underline = "".join(underline)
+        coords.insert(1, underline)
+        coords.append(underline)
+        summary += "\n".join(coords)
+        return summary
+
+    def getSlicesString(self):
+        point = [self.where_not_close[j][0] for j in range(len(self.where_not_close))]
+        start_idxs = []
+        end_idxs = []
+        slices = []
+        for i, s in enumerate(point):
+            start_idx = max(0, s - 3)
+            end_idx = min(self.npy_aie1.shape[i], s + 3)
+            start_idxs.append(start_idx)
+            end_idxs.append(end_idx)
+            slices.append(slice(start_idx, end_idx))
+
+        slices_string = ""
+        for i in range(len(start_idxs)):
+            slices_string += "{}:{},".format(start_idxs[i], end_idxs[i])
+        slices_string = slices_string[:-1]
+
+        summary = f"\n\n\nBaseline result in the slice range [{slices_string}]\n"
+        summary += getNpArrayString(self.npy_baseline1[tuple(slices)])
+
+        summary += f"\n\n\nAIE result in the slice range [{slices_string}]\n"
+        summary += getNpArrayString(self.npy_aie1[tuple(slices)])
+
+        summary += f"\n\n\nAbsolute difference in the slice range [{slices_string}]\n"
+        summary += getNpArrayString(self.abs_diff[tuple(slices)])
+        return summary
+
+    def getSummaryString(self, max_discrepancies_to_show):
+        summary = self.getBaseString()
+        summary += self.getDiscrepenciesString(max_discrepancies_to_show)
+        summary += self.getSlicesString()
+        return summary
+
+
+def compare(npy_baseline1, npy_aie1, rtol, atol, max_discrepancies_to_show=50):
```

Comment:
Why not use "npy_baseline" and "npy_aie"?

Reply:
I can't remember the history here, but I've dropped the '1's.